base/constants_syncdocs.go

```diff
@@ -457,10 +467,45 @@ func SetSyncInfoMetaVersion(ds DataStore, metaVersion string) error {
 	return err
 }
 
-// SerializeIfLonger returns name as a sha1 string if the length of the name is greater or equal to the length specificed. Otherwise, returns the original string.
+// SerializeIfLonger returns name as a sha1 string if the length of the name is greater or equal to the length specified. Otherwise, returns the original string.
 func SerializeIfLonger(name string, length int) string {
 	if len(name) < length {
 		return name
 	}
 	return Sha1HashString(name, "")
 }
+
+// CompareMetadataVersion Will build comparable build version for comparison with meta version defined in syncInfo, then
+// will return true if we require attachment migration, false if not.
+func CompareMetadataVersion(metaVersion string) (bool, error) {
+	var syncInfoVersion *ComparableBuildVersion
+	var err error
+	if metaVersion != "" {
+		syncInfoVersion, err = NewComparableBuildVersionFromString(metaVersion)
+		if err != nil {
+			return true, err
+		}
+	}
+	requiresAttachmentMigration, err := CheckRequireAttachmentMigration(syncInfoVersion)
+	if err != nil {
+		return true, err
+	}
+	return requiresAttachmentMigration, nil
+}
+
+// CheckRequireAttachmentMigration will return true if current metaVersion < 4.0.0, else false
+func CheckRequireAttachmentMigration(version *ComparableBuildVersion) (bool, error) {
+	if version == nil {
+		return true, nil
+	}
+	minVerStr := "4.0.0" // minimum meta version that needs to be defined for metadata migration. Any version less than this will require attachment migration
+	minVersion, err := NewComparableBuildVersionFromString(minVerStr)
+	if err != nil {
+		return true, nil
```

Comment:
We aren't passing the error here, but I think we should at least log this case with a warning and fail this case with cb_sg_devmode.